cli.js

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env node
+var fs = require('fs')
+var minimist = require('minimist')
+var fastq = require('./index')
+
+var argv = minimist(process.argv.slice(2), {
+  alias: {
+    help: 'h'
+  }
+})
+
+if (argv.help || argv._.length === 0) {
+  console.log(
+        'Usage: bionode-fastq <options> <fastq file [required]> <output file>\n' +
+        'If no output is provided, the result will be printed to stdout\n'
+    )
+  process.exit(0)
+}
+
+// if there is an output file present, write to the file. Otherwise write to stdout
+var output = argv._[1] ? fs.createWriteStream(argv._[1]) : process.stdout
+
+var fq = fastq.read(argv._[0])
+
+process.stdin.setEncoding('utf8')
+
+fq.on('data', function (data) {
```

Comment:
Ahh yes @Colelyman my bad - I thought this module was a stream. It does not make sense to pipe an event emitter. The CLI from this PR looks like it will work with the changes in refactoring branch - I suggest we merge this in now, finish refactoring (which I think is just some cleaning up and making sure tests pass), and as part of that can implement `fs.createReadStream(input).pipe(fastq)` in the CLI - which is what it looks like @tiagofilipe12 was [using](https://github.com/bionode/bionode-fastq/blob/a800d3fb533dd0b68786e084faf694a776aac952/lib/bionode-fastq.js#L60-L62) to test refactoring branch.